### PR TITLE
Remove unneeded code

### PR DESCRIPTION
**Pull request by ahmetmircik**
Remove unneeded code. Seen while working on https://github.com/hazelcast/hazelcast/pull/19511

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-lab-run

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystem.java [hunk 1]
```diff
@@ -236,11 +236,6 @@ public final void extendExpiryTime(Data dataKey, long now, long lastUpdateTime)
             return;
         }
 
-        Map<Data, ExpiryMetadata> expireTimeByKey = getOrCreateExpireTimeByKeyMap(false);
-        if (isEmpty()) {
-            return;
-        }
-
         ExpiryMetadata expiryMetadata = getExpiryMetadataForExpiryCheck(dataKey, expireTimeByKey);
```
Comment (ufukyilmaz) [inline hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystem.java:239]: How do we assert here that `expireTimeByKey` won't be null?

Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystem.java:239]: `isEmpty()` checks it.

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystem.java (hunk 1 above)
Comment (ufukyilmaz) [reply hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystem.java:239]: sorry for inconvenience, I missed the above occurrence.

**Review by ufukyilmaz — APPROVED** [review]
it looks good to me

**Pull request merged**